File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.go
func maxProfit(prices []int) int {
    
    var result int = 0
    
    minValue := prices[0]
    
    for i:=1; i < len(prices); i++ {
        
        if minValue > prices[i] {
            minValue = prices[i];
        } else {
            tmp := prices[i] - minValue
            if tmp > result {
                result = tmp
            }
        }
        
    }
    return result;
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        
  int minPrice = prices[0], result = 0;
        for (int i = 1; i < prices.length; i++) {

            if (prices[i] < minPrice) {
                minPrice = prices[i];
            } else {
                result = Math.max(result, prices[i] - minPrice);
            }


        }

        return result;
    }
}